### PR TITLE
TS-3743: Avoid crash due to accessing dead session under load

### PR DESCRIPTION
**Pull request by niq**
Patch by Sam Baskinger

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/674/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/778/ for details.

**Comment by zwoop** [comment]
Clang format i think

**Comment by SolidWallOfCode** [comment]
What is the use case for this? Can this happen with just HTTP/1 clients or only with HTTP/2?

**Comment by SolidWallOfCode** [comment]
Looking that the checks for `m_active` there is an apparently assumption that the value can change over the course of the `HttpSM::setup_internal_transfer`. Give the instance pointed at by `ua_session` should be locked along with the `HttpSM` instance, how can that happen? Even without locks any such change must happen in another thread in which case the member really needs to be `volatile`.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -6254,7 +6258,11 @@ HttpSM::setup_internal_transfer(HttpSMHandler handler_arg)
   tunnel.add_consumer(ua_entry->vc, HTTP_TUNNEL_STATIC_PRODUCER, &HttpSM::tunnel_handler_ua, HT_HTTP_CLIENT, "user agent");
 
   ua_entry->in_tunnel = true;
-  tunnel.tunnel_run(p);
+
+  // Only actually try to do this on live sessions.
+  if (ua_session->m_active) {
```
Comment (PSUdaemon) [inline proxy/http/HttpSM.cc]: Can we become active between line 6172 and line 6263?

Comment (PSUdaemon) [inline proxy/http/HttpSM.cc]: Oh, what @SolidWallOfCode said!

**Comment by bryancall** [comment]
Please clang format the change.

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
If this is still and issue please reopen.

**Pull request closed without merging**